articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+In the future, one such target machine will be a real quantum computer, but there are currently multiple simulators available, each with a particular use.
+For our purposes here, the most useful target machine is instance of the full-state simulator, `QuantumSimulator`, which calculates the program's behavior as if it were being executed on a noise-free quantum computer.
+
+So far, we've described what happens when a specific Q# operation is being executed.
+Regardless of whether Q# is used in a standalone application or with a host program, this general process is more or less the same---hence the QDK's flexibility.
+The differences between the methods therefore reveal themselves in *how* that Q# operation is called to be executed, and in what manner any results are returned.
+
+First, we discuss how this done with the Q# standalone application from the command line, and then proceed to using Python and C# host programs.
+We reserve the standalone application of Q# Jupyter Notebooks for last, because unlike the first three, it does not actually require a Q# file.
+
+
+## Q# from the command line
+One of the easiest ways to get started writing Q# programs is to avoid worrying about separate files and a second language altogether.
+Using Visual Studio Code or Visual Studio with the QDK extension allows for a seamless work flow in which we run Q# operations from only a single Q# file.
+
+For this, we will ultimately invoke the program's execution by entering
+```bash
+dotnet run
+```
+in the command line.
+Note that the terminal's directory location needs to be the same as the Q# file, so we recommend simply using the integrated terminal in VS Code or Visual Studio.
+
+### Add entry point to Q# file
+Of course, most Q# files will contain more than one operation.
+So, we need to let the compiler know *which* operation to execute when we provide the `dotnet run` command.
+
+This is done with a simple change to the Q# file itself: add a line with `@EntryPoint()` directly preceding the operation.
+
+Our file from above would therefore become
+```qsharp
+namespace NamespaceName {
+    open Microsoft.Quantum.Intrinsic;     // for the H operation
+    open Microsoft.Quantum.Measurement;   // for MResetZ
+
+	@EntryPoint()
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) { 
+            H(q);
+            return MResetZ(q);
+        }
+    }
+}
+```
+
+Now, a call of `dotnet run` from the command line leads to `MeasureSuperposition` being run, and the returned value is then printed directly to the terminal.
+So, you will see either `One` or `Zero` printed. 
+
+Note that it doesn't matter if you have more operations defined below it, only `MeasureSuperposition` will be run.
+Additionally, it's no problem if your operation includes [documentation comments](xref:microsoft.quantum.guide.filestructure#documentation-comments) before the operation declaration, the `@EntryPoint()` attribute can be simply placed above them.
```

Comment:
Has the term "attribute" been defined so far? We may want to explain it a little bit (perhaps in a sidebar), since the term "attribute" is .NET-y but we've used Python-ic or Java-like syntax.

Reply:
I referenced it as an attribute as that's what the release notes used (and the API ref for it gave me next-to-nothing), but personally I'm not too up on the .NET jargon. I'm nervous more explanation might distract from the page though, considering its utilitarian nature and the fact that many visitors might be here before even starting coding (don't want to hit them with too much jargon at one time). Would those details maybe fit elsewhere? Or if you think it's valuable here, just let me know.